### PR TITLE
Fix positional arguments

### LINKED ISSUE #35600: Trash with `enable_positional_arguments`
```
SET enable_positional_arguments = 1;
SELECT b FROM (SELECT 5 AS a, 'Hello' AS b ORDER BY a);

Received exception from server (version 22.4.1):
Code: 43. DB::Exception: Received from localhost:9000. DB::Exception: Positional argument out of bounds: 5 (exprected in range [1, 1]. (ILLEGAL_TYPE_OF_ARGUMENT)
```

### PR DESCRIPTION
**Pull request by kssenii**
Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix positional arguments with aliases. Closes https://github.com/ClickHouse/ClickHouse/issues/35600.

### CODE REVIEW

File: src/Interpreters/ExpressionAnalyzer.cpp [hunk 1]
```diff
@@ -100,20 +100,9 @@ bool checkPositionalArguments(ASTPtr & argument, const ASTSelectQuery * select_q
...
-            const auto * col_with_alias = dynamic_cast<const ASTWithAlias *>(column.get());
-            if (col_with_alias)
-            {
-                const auto & alias = col_with_alias->alias;
-                if (!alias.empty() && alias == group_by_expr_with_alias->alias)
-                    return false;
-            }
-        }
-    }
+    const auto * expr_with_alias = dynamic_cast<const ASTWithAlias *>(argument.get());
```
Comment (nickitat) [inline src/Interpreters/ExpressionAnalyzer.cpp:103]: what will be the result of this query `select b from (select 5 as a, 'Hello' as b order by 1) as t;`?

File: tests/queries/0_stateless/02006_test_positional_arguments.reference [hunk 2]
```diff
@@ -111,3 +111,7 @@ select substr('aaaaaaaaaaaaaa', 8) as a  group by a;
 aaaaaaa
 select substr('aaaaaaaaaaaaaa', 8) as a  group by substr('aaaaaaaaaaaaaa', 8);
 aaaaaaa
+select b from (select 5 as a, 'Hello' as b order by a);
+Hello
+select b from (select 5 as a, 'Hello' as b group by a);
```
Comment (nickitat) [inline tests/queries/0_stateless/02006_test_positional_arguments.reference:116]: we're going to test the same query twice, is it intentional?

Comment (kssenii) [reply tests/queries/0_stateless/02006_test_positional_arguments.reference:116]: group by / order by

Comment (nickitat) [reply tests/queries/0_stateless/02006_test_positional_arguments.reference:116]: sorry, my bad

**Review by nickitat — APPROVED** [review]
lgtm

**Comment by kssenii** [comment]
> Stateless tests (thread, actions) [3/3] — fail: 1

--  02177_issue_31009 Timeout dropping database test_pqcvv6 after test

> Stateless tests (memory, actions) [1/3] — fail: 1

-- 01532_primary_key_without_order_by_zookeeper -- not related

> Stateless tests (memory, actions) [1/3] — fail: 1

-- 02149_read_in_order_fixed_prefix -- not related

> Integration tests (release, actions) [1/2] — fail: 1

-- test_global_overcommit_tracker/test.py::test_overcommited_is_killed -- not related

**Pull request merged**